src/tools/compiletest/src/header.rs

```diff
@@ -395,7 +402,15 @@ impl TestProps {
                 );
                 config.set_name_directive(ln, STDERR_PER_BITWIDTH, &mut self.stderr_per_bitwidth);
                 config.set_name_directive(ln, INCREMENTAL, &mut self.incremental);
-                config.set_name_directive(ln, KNOWN_BUG, &mut self.known_bug);
+                config.set_name_value_directive(ln, KNOWN_BUG, &mut self.known_bug, |b| {
+                    match b.trim() {
+                        "unknown" => KnownBug::Unknown,
+                        bug if Regex::new(r".*#\d+").unwrap().is_match(bug) => {
```

Comment:
This should probably pull the regex out to a lazy_static cache (I think we have some examples in this file already?) to avoid re-compiling it. Alternatively, we might consider not using a regex here; split_once('#') and asserting the second part is a number would be equivalent and likely cheaper.

Reply:
Done.